core/infrastructure/imagestream.go

```diff
@@ -43,8 +44,10 @@ var imageStreamAnnotations = map[string]string{
 // ImageStreamHandler ...
 type ImageStreamHandler interface {
 	FetchImageStream(key types.NamespacedName) (*imgv1.ImageStream, error)
+	FetchTag(key types.NamespacedName, tag string) (*imgv1.ImageStreamTag, error)
 	MustFetchImageStream(key types.NamespacedName) (*imgv1.ImageStream, error)
-	CreateImageStream(name, namespace, imageName, tag string, addFromReference, insecureImageRegistry bool) *imgv1.ImageStream
+	CreateImageStreamIfNotExists(name, namespace, tag string, addFromReference bool, imageName string, insecureImageRegistry bool) (*imgv1.ImageStream, error)
```

Comment:
why not using the `types.NamespacedName` in this method ?

Reply:
done